### PR TITLE
nvwave.playWaveFile: async > asynchronous

### LINKED ISSUE #8607: nvwave.playWaveFile/Python 3 syntax: async is no longer accepted as a variable name, throws SyntaxError in Python 3.7
Hi,

Continuing work on researching Python 3 transition barriers:

### Background:
Python 3.5 introduces async and await keywords for working with asynchronous generators and other possibilities. However, this poses a syntax issue for NVDA: nvwave.playWaveFile has "async" keyword, which, when run with Python 3.7, produces SyntaxError.

### Steps to reproduce:
1. Run Python 2.7 and 3.7 interpreters.
2. Run the below code in both interpreters.

Code fragment:

async = True

### Actual behavior:
In 2.7, the code works, but in 3.7, SyntaxError is raised.

### Expected behavior:
Code should work in 3.7.

#### NVDA Installed/portable/running from source:
Not applicable

#### NVDA version:
Not applicable

#### Windows version:
Windows 10 Version 1803

#### Name and version of other software in use when reproducing the issue:
Python 2.7, 3.7

#### Other information about your system:
N/A

#### Does the issue still occur after restarting your PC?
Yes

#### Have you tried any other versions of NVDA?
N/A

### Possible solution:
One possible solution is to rename the keyword "async" to "_async" or "asynchronous". I would vote for the latter option as it is a bit clear as to what that keyword does.

### Impact:
Any module (including add-ons) that calls nvwave.playWaveFile with "async" set to True must now use the proposed keyword argument name, breaking compatibility with old NVDA releases.

Thanks.

### PR DESCRIPTION
**Pull request by josephsl**
### Link to issue number:
Fixes #8607
### Summary of the issue:
Change async argument in nvwave.playWaveFile to asynchronous.

### Description of how this pull request fixes the issue:
Python 3.5 introduces 'async' and 'await' keywords to deal with asynchronous generators and other possibilities. Since Python 3.7, use of these keywords as variable names is no longer allowed. In NVDA code, nvWave.playWaveFile is affected, so rename 'async' to 'asynchronous'.

### Testing performed:
Compiled and tested various code paths that calls nvwave.playWaveFile with the new keyword arguments.

### Known issues with pull request:
Any add-on that sets "async" argument to True will be affected.

### Change log entry:
Changes for developers:
The "async" argument in nvwave.playWaveFile has been renamed to "asynchronous". Add-ons that relied on this name should be modified to use the new argument name. (#8607)

### CODE REVIEW

File: source/nvwave.py
Comment (leonardder) [inline source/nvwave.py]: May be add information about parameters to the doc string while at it?

**Review by LeonarddeR — DISMISSED** [review]

**Comment by josephsl** [comment]
Hi, shall be done. Thanks for catching this.

**Comment by michaelDCurran** [comment]
Hold off until @feerrenrut has finished add-on versioning.

**Comment by feerrenrut** [comment]
I've changed the base branch on this back to master, since it doesn't break compatibility with python 2, or introduce any risky changes there. Merging to master reduces the number of things that may require merging. 

Thinking about risk, is it possible for this name change to affect add-ons?

**Comment by josephsl** [comment]
Hi, not that I know of. Thanks.

**Comment by leonardder** [comment]
[**@feerrenrut**](https://github.com/feerrenrut) commented on [13 May 2019, 19:11 CEST](https://github.com/nvaccess/nvda/pull/8647#issuecomment-491908130 "2019-05-13T17:11:03Z - Replied by Github Reply Comments"):
> ... it doesn't break compatibility with python 2, or introduce any risky changes there. Merging to master reduces the number of things that may require merging.

Note that all code that passes async as a keyword argument will fail after merging this. While I can't think of any add-ons that utilize this code, I'd be very reluctant to merge this into a Python 2 version of NVDA if i wanted to make sure that i wouldn't break things. Note that there's nothing wrong with breaking things like this if we agree that we don't care.

**Comment by feerrenrut** [comment]
> I'd be very reluctant to merge this into a Python 2 version of NVDA if i wanted to make sure that i wouldn't break things

Yes, I think this introduces too much unnecessary risk. Python 3 migration is going to break add-ons anyway, so we will direct this risk there. I will re-target this PR.

**Comment by josephsl** [comment]
Hi, yep, thanks for rebasing it (I’ll merge in Threshold then). Thanks.

**Comment by josephsl** [comment]
Hi,

Conflict resolved and is ready for another round of review.

Steps taken:

1. Did grep -lr "async" and located instances of nvwave call.
2. Changed to "asynchronous".

Thanks.

**Review by michaelDCurran — APPROVED** [review]

**Pull request merged**
